doc/crypto/CT_POLICY_EVAL_CTX.pod

```diff
@@ -0,0 +1,84 @@
+=pod
+
+=head1 NAME
+
+CT_POLICY_EVAL_CTX -
+Encapsulates the data required to evaluate whether SCTs meet a Certificate Transparency policy
+
+=head1 SYNOPSIS
+
+ #include <openssl/ct.h>
+
+ CT_POLICY_EVAL_CTX *CT_POLICY_EVAL_CTX_new(void);
+ void CT_POLICY_EVAL_CTX_free(CT_POLICY_EVAL_CTX *ctx);
+ X509* CT_POLICY_EVAL_CTX_get0_cert(const CT_POLICY_EVAL_CTX *ctx);
+ void CT_POLICY_EVAL_CTX_set0_cert(CT_POLICY_EVAL_CTX *ctx, X509 *cert);
+ X509* CT_POLICY_EVAL_CTX_get0_issuer(const CT_POLICY_EVAL_CTX *ctx);
+ void CT_POLICY_EVAL_CTX_set0_issuer(CT_POLICY_EVAL_CTX *ctx, X509 *issuer);
+ const CTLOG_STORE *CT_POLICY_EVAL_CTX_get0_log_store(const CT_POLICY_EVAL_CTX *ctx);
+ void CT_POLICY_EVAL_CTX_set0_log_store(CT_POLICY_EVAL_CTX *ctx, CTLOG_STORE *log_store);
+
+=head1 DESCRIPTION
+
+A B<CT_POLICY_EVAL_CTX> is used by functions that evaluate whether Signed
+Certificate Timestamps (SCTs) fulfil a Certificate Transparency (CT) policy.
+This policy may be, for example, that at least one valid SCT is available. To
+determine this, an SCT's signature must be verified. This requires:
+
+=over
+
+=item * the public key of the log that issued the SCT
+
+=item * the certificate that the SCT was issued for
+
+=item * the issuer certificate (if the SCT was issued for a pre-certificate)
+
+=back
+
+The above requirements are met using the setters described below.
+
+B<CT_POLICY_EVAL_CTX_new>() creates an empty policy evaluation context. This
+should then be populated using:
+
+=over
+
+=item * B<CT_POLICY_EVAL_CTX_set0_cert>() to provide the certificate the SCTs were issued for
+
+=item * B<CT_POLICY_EVAL_CTX_set0_issuer>() to provide the issuer certificate
+
+=item * B<CT_POLICY_EVAL_CTX_set0_log_store>() to provide a list of logs that are trusted as sources of SCTs
+
+=back
+
+None of these setters take ownership of the pointers passed to them.
```

Comment:
(Drive-by.) Sadly the sister X509 API mixes `set0` semantics in weird ways (see https://github.com/openssl/openssl/blob/master/crypto/x509/x509_vfy.c) but basically, an X509_STORE_CTX is one-shot and holds temporary references to structs, while an X509_STORE is long-lived. Whether the X509_STORE_CTX references are unique or shared basically seems to depend on whether the struct is created by the verify routine or elsewhere in the handshake... But yes, agreed, we should rather be consistent in our naming from now on.

Back to the very similar CT structs: it should be pretty typical here that CTLOG_STORE outlives the X509 which in turn outlives the one-shot CT_POLICY_EVAL_CTX. So, dup'ing the structures seems wasteful, plus I'm not even sure what the side effects would be with all the extension caching and whatnot going on in the X509 - I do not think that anything else in the handshake dup's the X509 so far?? Correctly `set1`-refcounting the certs seems like a fine idea, though. (Even though OpenSSL also mixes `set1` for shared refcounting and grabbing an independent copy...)

Dup'ing the long-lived, practically static, large CTLOG_STORE is even more wrong - there, I think I'd actually be fine with a `_set_shared_CTLOG_STORE` and a warning that the store must outlive the ctx, though refcounting is cleaner I guess...